src/libraries/Microsoft.PowerFx.Core/Public/Values/PagedRecordValue.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Public.Values
+{
+    // PagedRecordValue is a RecordValue that can be paged through.
+    // It is composed of the "nextLink" string and a table with the results
+    internal class PagedRecordValue : RecordValue
+    {
+        public RecordValue CurrentPage { get; private set; }
+
+        // Method used to retrieve the next page
+        public Func<FormulaValue> GetNextRecord { get; private set; }
+
+        public int MaxRows { get; private set; }
+
+        public PagedRecordValue(RecordValue recordValue, Func<FormulaValue> getNextRecord, int maxRows)
+            : base(recordValue.Type)
+        {
+            CurrentPage = recordValue;
+            GetNextRecord = getNextRecord;
+            MaxRows = maxRows;
+        }
+
+        protected internal override bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result)
+        {
+            bool b = CurrentPage.TryGetField(fieldType, fieldName, out result);
+
+            // If we retrieve the table (via "value" field), we need to wrap it in a PagedTableValue
+            if (fieldType is TableType && result is TableValue tv)
+            {
+                result = new PagedTableValue(tv, GetNextRecord, fieldType, fieldName, MaxRows);
+            }
+
+            return b;
+        }
+    }
+
+    internal class PagedTableValue : TableValue
+    {
+        public TableValue CurrentTable { get; private set; }
+
+        public Func<FormulaValue> GetNextRecord { get; private set; }
+
+        public FormulaType FieldType { get; private set; }
+
+        public string FieldName { get; private set; }
```

Comment:
Can you update name to be clear which field these refer to?   The next link's, or the value? 

If it's the value, can FieldType be TableType?

Reply:
FieldType is a TableType but not necessarily = TableType